linera-core/src/worker.rs

```diff
@@ -1112,7 +1122,23 @@ where
         );
         self.storage.clock().sleep_until(block.timestamp).await;
         let local_time = self.storage.clock().current_time();
-        let outcome = chain.execute_block(block, local_time).await?;
```

Comment:
oh wow we were re-executing valid blocks?

Reply:
(In the case where they were re-proposed in a later round, yes.)